internal/service/alarm_mapper.go

```diff
@@ -0,0 +1,332 @@
+/*
+Copyright 2023 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+	neturl "net/url"
+
+	"github.com/imdario/mergo"
+	"github.com/openshift-kni/oran-o2ims/internal/data"
+	"github.com/openshift-kni/oran-o2ims/internal/jq"
+	"github.com/openshift-kni/oran-o2ims/internal/k8s"
+
+	"github.com/itchyny/gojq"
+)
+
+type AlarmMapper struct {
+	logger              *slog.Logger
+	backendClient       *http.Client
+	resourceServerURL   string
+	resourceServerToken string
+	extensions          []string
+	jqTool              *jq.Tool
+}
+
+// AlarmMapperBuilder contains the data and logic needed to create a new AlarmMapper.
+type AlarmMapperBuilder struct {
+	logger        *slog.Logger
+	backendClient *http.Client
+	//transportWrapper    func(http.RoundTripper) http.RoundTripper
+	resourceServerURL   string
+	resourceServerToken string
+	extensions          []string
+}
+
+// AlarmMapper creates a builder that can then be used to configure
+// and create a handler for the AlarmMapper.
```

Comment:
Review the comments: this isn't accurate, it talks about a "handler" but there is no handler here.

Reply:
Right. Can be something like: NewAlarmMapper creates a builder that can be used to create a mapper of Alertmanager alerts to AlarmEventRecord objects.